packages/editor/src/store/private-actions.js

```diff
@@ -109,3 +110,113 @@ export const hideBlockTypes =
 			.dispatch( preferencesStore )
 			.set( 'core', 'hiddenBlockTypes', [ ...mergedBlockNames ] );
 	};
+
+/**
+ * Save entity records marked as dirty.
+ *
+ * @param {Object}   options                      Options for the action.
+ * @param {Function} [options.onSave]             Callback when saving happens.
+ * @param {object[]} [options.dirtyEntityRecords] Array of dirty entities.
+ * @param {object[]} [options.entitiesToSkip]     Array of entities to skip saving.
+ * @param {Function} [options.close]              Callback when the actions is called. It should be consolidated with `onSave`.
+ */
+export const saveDirtyEntities =
+	( { onSave, dirtyEntityRecords = [], entitiesToSkip = [], close } = {} ) =>
```

Comment:
The callback argument seems a bit odd to me. This is a pattern I've rarely seen with store actions. Do you think consumer components can handle them? Also, consumers should have all the data to generate the `entitiesToSave` array.

Reply:
For sure `close` should be consolidated probably with the `onSave` and also see what will need more changing, but I was thinking of doing it separately as this is already moving a lots of stuff around and this action is private.

The end goal is the consolidation of publishing flows between editors, and this in essence is a first step towards that, by removing an extra `save` button.

Reply:
Makes sense. Let's include cleanup/refactoring in the follow-up PR(s).